python_modules/dagster/dagster/_core/external_execution/resource.py

```diff
@@ -1,23 +1,14 @@
-from typing import Dict, Mapping, Optional, Sequence, Union
+from abc import ABC, abstractmethod
+from typing import Optional
 
 from dagster_external.protocol import ExternalExecutionExtras, ExternalExecutionIOMode
 from pydantic import Field
 
 from dagster._config.pythonic_config import ConfigurableResource
 from dagster._core.execution.context.compute import OpExecutionContext
-from dagster._core.external_execution.task import (
-    ExternalExecutionTask,
-)
 
 
-class ExternalExecutionResource(ConfigurableResource):
-    env: Optional[Dict[str, str]] = Field(
-        default=None,
-        description="An optional dict of environment variables to pass to the subprocess.",
-    )
-    cwd: Optional[str] = Field(
-        default=None, description="Working directory in which to launch the subprocess command."
-    )
+class ExternalExecutionResource(ConfigurableResource, ABC):
     input_mode: ExternalExecutionIOMode = Field(default="stdio")
     output_mode: ExternalExecutionIOMode = Field(default="stdio")
```

Comment:
the inclusion of modes on this generic class is upping my priority of getting rid of the concept entirely